### ISSUE TITLE
Hook is not working

### ISSUE THREAD

**Issue by rallek** [zikula-modules/MediaModule#168]
I have enabled the collection UI hook in News. When opening a news for editing I do get the following:

```
TypeError:
Return value of Zikula\Bundle\HookBundle\Hook\DisplayHook::getUrl() must be an instance of Zikula\Bundle\CoreBundle\UrlInterface, null returned

  at vendor/zikula/hook-bundle/Hook/DisplayHook.php:70
  at Zikula\Bundle\HookBundle\Hook\DisplayHook->getUrl()
     (src/extensions/cmfcmf/mediamodule/Entity/HookedObject/HookedObjectEntity.php:119)
  at Cmfcmf\Module\MediaModule\Entity\HookedObject\HookedObjectEntity->__construct(object(DisplayHook))
     (src/extensions/cmfcmf/mediamodule/Entity/HookedObject/Repository/HookedObjectRepository.php:44)
  at Cmfcmf\Module\MediaModule\Entity\HookedObject\Repository\HookedObjectRepository->getByHookOrCreate(object(DisplayHook))
     (src/extensions/cmfcmf/mediamodule/HookProvider/CollectionUiHooksProvider.php:90)
  at Cmfcmf\Module\MediaModule\HookProvider\CollectionUiHooksProvider->uiEdit(object(DisplayHook), 'munewsmodule.ui_hooks.messages.form_edit', object(TraceableEventDispatcher))
     (vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/WrappedListener.php:117)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(DisplayHook), 'munewsmodule.ui_hooks.messages.form_edit', object(TraceableEventDispatcher))
     (vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener)), 'munewsmodule.ui_hooks.messages.form_edit', object(DisplayHook))
     (vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(DisplayHook), 'munewsmodule.ui_hooks.messages.form_edit')
     (vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php:151)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(DisplayHook), 'munewsmodule.ui_hooks.messages.form_edit')
     (vendor/zikula/hook-bundle/Dispatcher/HookDispatcher.php:51)
  at Zikula\Bundle\HookBundle\Dispatcher\HookDispatcher->dispatch('munewsmodule.ui_hooks.messages.form_edit', object(DisplayHook))
     (vendor/zikula/hook-bundle/Twig/Extension/HookExtension.php:75)
  at Zikula\Bundle\HookBundle\Twig\Extension\HookExtension->notifyDisplayHooks('munewsmodule.ui_hooks.messages.form_edit', 1, null, true)
     (vendor/twig/twig/src/Environment.php(358) : eval()'d code:659)
  at __TwigTemplate_befac7b40543c166a99ac8512684a05061a15d367b2ebba978494ceb5f34bd2e->block_content(array('routeArea' => 'admin', 'thumbRuntimeOptions' => array('messageImageUpload1' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload2' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload3' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload4' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null))), 'relationThumbRuntimeOptions' => array('thumbnail' => array('size' => array(100, 75), 'mode' => 1, 'extension' => null)), 'permissionHelper' => object(PermissionHelper), 'featureActivationHelper' => object(FeatureActivationHelper), 'inlineUsage' => false, 'mode' => 'edit', 'translationsEnabled' => false, 'supportedLanguages' => array('de', 'en'), 'actions' => array('update' => array('id' => 'update', 'title' => 'Aktualisieren', 'buttonClass' => 'btn-success'), 'archive' => array('id' => 'archive', 'title' => 'Archiv', 'buttonClass' => ''), 'delete' => array('id' => 'delete', 'title' => 'Löschen', 'buttonClass' => 'btn-danger'), 'unpublish' => array('id' => 'unpublish', 'title' => 'Veröffentlichung aufheben', 'buttonClass' => ''), 'trash' => array('id' => 'trash', 'title' => 'Papierkorb', 'buttonClass' => '')), 'formHookTemplates' => array(), 'form' => object(FormView), 'message' => object(MessageEntity), 'supportsHookSubscribers' => true, 'app' => object(AppVariable), 'pagevars' => object(ParameterBag), 'themevars' => object(ParameterBag), 'currentUser' => object(CurrentUserApi), 'baseTemplate' => 'adminBase', 'language' => 'de', 'imageUpload' => false, 'hookId' => 1), array('header' => array(object(__TwigTemplate_53806ba3c68fc4360830d684224ec0a2c8ccdf76c4b36eee3fd60c503eee2955), 'block_header'), 'appTitle' => array(object(__TwigTemplate_3fbc232666c81044022d16beb813b3b782a0938e04faaf6c89598105f3e9efd6), 'block_appTitle'), 'titleArea' => array(object(__TwigTemplate_3fbc232666c81044022d16beb813b3b782a0938e04faaf6c89598105f3e9efd6), 'block_titleArea'), 'title' => array(object(__TwigTemplate_befac7b40543c166a99ac8512684a05061a15d367b2ebba978494ceb5f34bd2e), 'block_title'), 'content' => array(object(__TwigTemplate_befac7b40543c166a99ac8512684a05061a15d367b2ebba978494ceb5f34bd2e), 'block_content'), 'footer' => array(object(__TwigTemplate_befac7b40543c166a99ac8512684a05061a15d367b2ebba978494ceb5f34bd2e), 'block_footer'), 'assets' => array(object(__TwigTemplate_53806ba3c68fc4360830d684224ec0a2c8ccdf76c4b36eee3fd60c503eee2955), 'block_assets'), 'admin_page_icon' => array(object(__TwigTemplate_befac7b40543c166a99ac8512684a05061a15d367b2ebba978494ceb5f34bd2e), 'block_admin_page_icon')))
     (vendor/twig/twig/src/Template.php:173)
  at Twig\Template->displayBlock('content', array('routeArea' => 'admin', 'thumbRuntimeOptions' => array('messageImageUpload1' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload2' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload3' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null)), 'messageImageUpload4' => array('thumbnail' => array('size' => array(240, 180), 'mode' => 'inset', 'extension' => null))), 'relationThumbRuntimeOptions' => array('thumbnail' => array('size' => array(100, 75), 'mode' => 1, 'extension' => null)), 'permissionHelper' => object(PermissionHelper), 'featureActivationHelper' => object(FeatureActivationHelper), 'inlineUsage' => false, 'mode' => 'edit', 'translationsEnabled' => false, 'supportedLanguages' => array('de', 'en'), 'actions' => array('update' => array('id' => 'update', 'title' => 'Aktualisieren', 'buttonClass' => 'btn-success'), 'archive' => array('id' => 'archive', 'title' => 'Archiv', 'buttonClass' => ''), 'delete' => array('id' => 'delete', 'title' => 'Löschen', 'buttonClass' => 'btn-danger'), 'unpublish' => array('id' => 'unpublish', 'title' => 'Veröffentlichung aufheben', 'buttonClass' => ''), 'trash' => array('id' => 'trash', 'title' => 'Papierkorb', 'buttonClass' => '')), 'formHookTemplates' => array(), 'form' => object(FormView), 'message' => object(MessageEntity), 'supportsHookSubscribers' => true, 'app' => object(AppVariable), 'pagevars' => object(ParameterBag), 'themevars' => object(ParameterBag), 'currentUser' => object(CurrentUserApi), 'baseTemplate' => 'adminBase'), array('header' => 
```

**Comment by Guite** [comment]
Working on this next. It's not a media issue.

**Comment by Guite** [comment]
Fixed in 95d9e81199c994d50809e5a3aa2bd3ce2afde54f

But hooks do not work yet -> continued in #175
